charts/prefect-server/templates/background-services-deployment.yaml

```diff
@@ -0,0 +1,156 @@
+{{- if .Values.backgroundServices.runAsSeparateDeployment }}
```

Comment:
Just a thought here: given how much of this file is identical to `server-deployment.yaml`, we could consider putting the config in a single file as a template object and then including that template object with the related overrides in each unique case.

As an example:
- [statefulset common `spec`](https://gitlab.com/gitlab-org/charts/gitlab/-/blob/master/charts/gitlab/charts/gitaly/templates/_statefulset_spec.yaml?ref_type=heads)
- [statefulset declaration referencing the common `spec`](https://gitlab.com/gitlab-org/charts/gitlab/-/blob/master/charts/gitlab/charts/gitaly/templates/statefulset.yml?ref_type=heads)

This can just be an optimization we make later, it just stood out to me as a ~150 line file that's mostly the same content 👍🏼